src/shared/types/checkReactTypeSpec.js

```diff
@@ -34,101 +28,44 @@ if (
-      } catch (ex) {
-        error = ex;
+  function formatMessage(message) {
+    if (__DEV__) {
+      let componentStackInfo = '';
+      if (!ReactComponentTreeHook) {
+        ReactComponentTreeHook = require('ReactComponentTreeHook');
       }
-      warning(
-        !error || error instanceof Error,
-        '%s: type specification of %s `%s` is invalid; the type checker ' +
-        'function must return `null` or an `Error` but returned a %s. ' +
-        'You may have forgotten to pass an argument to the type checker ' +
-        'creator (arrayOf, instanceOf, objectOf, oneOf, oneOfType, and ' +
-        'shape all require an argument).',
-        componentName || 'React class',
-        ReactPropTypeLocationNames[location],
-        typeSpecName,
-        typeof error
-      );
-      if (error instanceof Error && !(error.message in loggedTypeFailures)) {
-        // Only monitor this failure once because there tends to be a lot of the
-        // same error.
-        loggedTypeFailures[error.message] = true;
-
-        var componentStackInfo = '';
-
-        if (__DEV__) {
-          if (!ReactComponentTreeHook) {
-            ReactComponentTreeHook = require('ReactComponentTreeHook');
-          }
-          if (workInProgressOrDebugID != null) {
-            if (typeof workInProgressOrDebugID === 'number') {
-              // DebugID from Stack.
-              const debugID = workInProgressOrDebugID;
-              componentStackInfo = ReactComponentTreeHook.getStackAddendumByID(debugID);
-            } else if (typeof workInProgressOrDebugID.tag === 'number') {
-              // This is a Fiber.
-              // The stack will only be correct if this is a work in progress
-              // version and we're calling it during reconciliation.
-              const workInProgress = workInProgressOrDebugID;
-              componentStackInfo = ReactComponentTreeHook.getStackAddendumByWorkInProgressFiber(workInProgress);
-            }
-          } else if (element !== null) {
-            componentStackInfo = ReactComponentTreeHook.getCurrentStackAddendum(element);
-          }
+      if (workInProgressOrDebugID != null) {
+        if (typeof workInProgressOrDebugID === 'number') {
+          // DebugID from Stack.
+          const debugID = workInProgressOrDebugID;
+          componentStackInfo = ReactComponentTreeHook.getStackAddendumByID(debugID);
+        } else if (typeof workInProgressOrDebugID.tag === 'number') {
+          // This is a Fiber.
+          // The stack will only be correct if this is a work in progress
+          // version and we're calling it during reconciliation.
+          const workInProgress = workInProgressOrDebugID;
+          componentStackInfo = ReactComponentTreeHook.getStackAddendumByWorkInProgressFiber(workInProgress);
         }
-
-        warning(
-          false,
-          'Failed %s type: %s%s',
-          location,
-          error.message,
-          componentStackInfo
```

Comment:
At FB we actually split each of these into a separate column in the database logs. So if we can preserve that, that'd be great.